modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -2466,6 +2365,100 @@ private void ackNodeMetrics(DecimalFormat dblFmt,
+
+                long offHeapUsed = region.pageMemory().systemPageSize() * pagesCnt;
+                long offHeapInit = regCfg.getInitialSize();
+                long offHeapMax = regCfg.getMaxSize();
+                long offHeapComm = region.memoryMetrics().getOffHeapSize();
+
+                long offHeapUsedInMBytes = offHeapUsed / MEGABYTE;
+                long offHeapMaxInMBytes = offHeapMax / MEGABYTE;
+                long offHeapCommInMBytes = offHeapComm / MEGABYTE;
+                long offHeapInitInMBytes = offHeapInit / MEGABYTE;
+
+                double freeOffHeapPct = offHeapMax > 0 ?
+                    ((double)((offHeapMax - offHeapUsed) * 100)) / offHeapMax : -1;
+
+                offHeapUsedSummary += offHeapUsed;
+                offHeapMaxSummary += offHeapMax;
+                offHeapCommSummary += offHeapComm;
+                loadedPages += pagesCnt;
+
+                String type = "user";
+
+                try {
+                    if (region == database.dataRegion(null))
+                        type = "default";
+                    else if (INTERNAL_DATA_REGION_NAMES.contains(regCfg.getName()))
+                        type = "internal";
+                }
+                catch (IgniteCheckedException ice) {
+                    // Should never happen
+                    ice.printStackTrace();
+                }
+
+                dataRegionsInfo.a("    ^--   ")
+                    .a(regCfg.getName()).a(" region [type=").a(type)
+                    .a(", persistence=" + regCfg.isPersistenceEnabled())
+                    .a(", lazyAlloc=" + regCfg.isLazyMemoryAllocation()).a(',').a(NL)
+                    .a("      ...  ")
+                    .a("initCfg=").a(dblFmt.format(offHeapInitInMBytes))
+                    .a("MB, maxCfg=").a(dblFmt.format(offHeapMaxInMBytes))
+                    .a("MB, usedRam=").a(dblFmt.format(offHeapUsedInMBytes))
+                    .a("MB, freeRam=").a(dblFmt.format(freeOffHeapPct))
+                    .a("%, allocRam=").a(dblFmt.format(offHeapCommInMBytes)).a("MB");
+
+                if (regCfg.isPersistenceEnabled()) {
+                    long pdsUsed = region.memoryMetrics().getTotalAllocatedSize();
+                    long pdsUsedInMBytes = pdsUsed / MEGABYTE;
+
+                    pdsUsedSummary += pdsUsed;
+
+                    dataRegionsInfo.a(", allocTotal=").a(dblFmt.format(pdsUsedInMBytes)).a("MB");
+
+                    persistenceEnabled = true;
+                }
+
+                dataRegionsInfo.a(']').a(NL);
+            }
+        }
+
+        long offHeapUsedInMBytes = offHeapUsedSummary / MEGABYTE;
+        long offHeapCommInMBytes = offHeapCommSummary / MEGABYTE;
+        long pdsUsedMBytes = pdsUsedSummary / MEGABYTE;
+
+        double freeOffHeapPct = offHeapMaxSummary > 0 ?
+            ((double)((offHeapMaxSummary - offHeapUsedSummary) * 100)) / offHeapMaxSummary : -1;
+
+        return (includeMemoryStatistics ?
```

Comment:
It is hard for me to read ternary operator for this statement. I suggest to extract methods assembling message pieces.